src/order/basic.lean

```diff
@@ -612,6 +614,37 @@ or_iff_not_imp_left.2 $ λ h,
   ⟨λ a ha₁, le_of_not_gt $ λ ha₂, h ⟨a, ha₁, ha₂⟩,
     λ a ha₂, le_of_not_gt $ λ ha₁, h ⟨a, ha₁, ha₂⟩⟩
 
+/-- Order equipped with a sensible successor function. -/
+class succ_order (α : Type u) [preorder α] :=
+(succ : α → α)
+(lt_iff_succ_le : ∀ a b, a < b ↔ succ a ≤ b)
+(lt_succ_iff_le : ∀ a b, a < succ b ↔ a ≤ b)
```

Comment:
In this case, `fin (n + 1)` wouldn't be a `succ_order` since there is no `succ` for `n`, right? Do you have an idea for a "`maybe_succ_order`" that also includes `fin`?